test-host/src/androidTest/java/com/treasuredata/android/TreasureDataInstrumentTest.java

```diff
@@ -0,0 +1,135 @@
+package com.treasuredata.android;
+
+import android.app.Application;
+import android.content.Context;
+import android.support.test.InstrumentationRegistry;
+
+import junit.framework.TestCase;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import io.keen.client.java.KeenCallback;
+import io.keen.client.java.KeenProject;
+
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.spy;
+
+public class TreasureDataInstrumentTest extends TestCase {
+    private static final String DUMMY_API_KEY = "dummy_api_key";
+
+    class Event {
+        String tag;
+        Map<String, Object> event;
+        Event(String tag, Map<String, Object>event) {
+            this.tag = tag;
+            this.event = event;
+        }
+    }
+
+    class MockTDClient extends TDClient {
+        Exception exceptionOnQueueEventCalled;
```

Comment:
These exceptions seem to have no use currently?

Reply:
hmmm. Not sure what you meant? It's used inside `queueEvent`

Reply:
I meant when will this ever have any value other than null?
But now I see it gets copied over from `TreasureDataTest`, that explains it.